application/src/main/java/org/togetherjava/tjbot/features/basic/Starboard.java

```diff
@@ -20,8 +30,17 @@ public Starboard(Config config) {
     @Override
     public void onMessageReactionAdd(@NotNull MessageReactionAddEvent event) {
         String emojiName = event.getReaction().getEmoji().asCustom().getName();
-        MessageChannel channel = event.getChannel();
-        // TODO
-
+        Guild guild = event.getGuild();
+        GuildChannel channel = event.getGuildChannel();
+        if (!config.getEmojiNames().contains(emojiName) || !guild.getPublicRole().hasPermission(channel, Permission.VIEW_CHANNEL)) {
```

Comment:
i feel like this either needs a comment or should be moved into a well named method, so that its clear why we do this check